languages/rust/template/src/bin/pipe.rs

```diff
@@ -0,0 +1,165 @@
+// This file provides interop with the langserver
+//
+use algorithm;
+
+use base64;
+use serde::{Deserialize, Serialize};
+use serde_json;
+
+use algorithmia::error::{Error, ErrorKind, ResultExt};
+use algorithmia::prelude::{AlgoIo, EntryPoint};
+use serde_json::Value;
+use std::error::Error as StdError;
+use std::fs::OpenOptions;
+use std::io::{self, BufRead, Write};
+use std::process;
+
+const ALGOOUT: &'static str = "/tmp/algoout";
+
+#[derive(Deserialize)]
+struct Request {
+    data: Value,
+    content_type: String,
+}
+
+#[derive(Serialize)]
+struct AlgoSuccess {
+    result: Value,
+    metadata: RunnerMetadata,
+}
+
+#[derive(Serialize)]
+struct AlgoFailure {
+    error: RunnerError,
+}
+
+#[derive(Serialize)]
+struct RunnerMetadata {
+    content_type: String,
+}
+
+#[derive(Serialize)]
+struct RunnerError {
+    message: String,
+    error_type: &'static str,
+}
+
+impl AlgoSuccess {
+    fn new<S: Into<String>>(result: Value, content_type: S) -> AlgoSuccess {
+        AlgoSuccess {
+            result: result,
+            metadata: RunnerMetadata {
+                content_type: content_type.into(),
+            },
+        }
+    }
+}
+
+impl AlgoFailure {
+    fn new(err: &dyn StdError) -> AlgoFailure {
+        AlgoFailure {
+            error: RunnerError {
+                message: error_cause_chain(err),
+                error_type: "AlgorithmError",
+            },
+        }
+    }
+
+    fn system(err: &dyn StdError) -> AlgoFailure {
+        AlgoFailure {
+            error: RunnerError {
+                message: error_cause_chain(err),
+                error_type: "SystemError",
+            },
+        }
+    }
+}
+
+fn main() {
+    let mut algo = algorithm::Algo::default();
```

Comment:
Where is this defined? Is this using reflection? I wasn't aware that was possible.

Reply:
`Cargo.toml` specifies that the lib (`src/lib.rs`) be compiled using the name `algorithm`. This has been the pattern for rust algos since the beginning.

Reply:
this no longer exists after latest commit.